Sources/MUXSDKStatsInternal/Task+Timeout.swift

```diff
@@ -4,18 +4,26 @@ struct TimeoutError: Error {
 }
 
 @available(iOS 13, tvOS 13, *)
-func withTimeout<Output>(of seconds: TimeInterval, operation: sending @escaping @isolated(any) () async throws -> Output) async throws -> Output {
-    try await withThrowingTaskGroup(of: Output.self) { group in
-        group.addTask(operation: operation)
-        group.addTask {
-            try await Task.sleep(nanoseconds: UInt64(seconds * Double(NSEC_PER_SEC)))
-            throw TimeoutError()
-        }
-
-        defer {
-            group.cancelAll()
+func withTimeout<T: Sendable>(seconds: TimeInterval, operation: sending @escaping @isolated(any) () async throws -> T) async throws -> T {
+    let workTask = Task {
+        try await operation()
+    }
+    return try await withCheckedThrowingContinuation { continuation in
+        Task {
+            await withThrowingTaskGroup(of: T.self) { group in
+                defer {
+                    workTask.cancel()
+                }
+                group.addTask {
+                    try await workTask.value
+                }
+                group.addTask {
+                    try await Task.sleep(nanoseconds: UInt64(seconds * Double(NSEC_PER_SEC)))
+                    throw TimeoutError()
+                }
+                let firstResult = await group.nextResult()!
+                continuation.resume(with: firstResult)
```

Comment:
`withCheckedThrowingContinuation` can now return immediately. `withThrowingTaskGroup` always waits until all subtasks return, which means waiting until they successfully notice cancellation and clean up. See tests added for blocking operations and never-returning operations!